File: system-test/java/src/main/java/com/optivem/shop/testkit/core/usecase/shop/usecases/GoToShop.java
package com.optivem.shop.testkit.core.usecase.shop.usecases;

import com.optivem.shop.testkit.driver.port.shop.ShopDriver;
import com.optivem.shop.testkit.core.usecase.shop.usecases.base.BaseShopUseCase;
import com.optivem.shop.testkit.core.shared.UseCaseResult;
import com.optivem.shop.testkit.core.shared.UseCaseContext;
import com.optivem.shop.testkit.core.shared.VoidVerification;

public class GoToShop extends BaseShopUseCase<Void, VoidVerification> {
    public GoToShop(ShopDriver driver, UseCaseContext context) {
        super(driver, context);
    }

    @Override
    public UseCaseResult<Void, VoidVerification> execute() {
        var result = driver.goToShop();
        return new UseCaseResult<>(result, context, VoidVerification::new);
    }
}

File: system-test/typescript/src/testkit/dsl/core/usecase/shop/usecases/GoToShop.ts
import type { ShopDriver } from '../../../../../driver/port/shop/shop-driver.js';
import { UseCaseResult } from '../../../shared/use-case-result.js';
import { VoidVerification } from '../../../shared/void-verification.js';
import type { UseCaseContext } from '../../../shared/use-case-context.js';
import { BaseShopUseCase } from './base/BaseShopUseCase.js';

export class GoToShop extends BaseShopUseCase<void, VoidVerification> {
  constructor(driver: ShopDriver, context: UseCaseContext) {
    super(driver, context);
  }

  async execute(): Promise<UseCaseResult<void, VoidVerification>> {
    const result = await this.driver.goToShop();

    return new UseCaseResult(
      result,
      this.context,
      (_, ctx) => new VoidVerification(undefined, ctx),
    );
  }
}
